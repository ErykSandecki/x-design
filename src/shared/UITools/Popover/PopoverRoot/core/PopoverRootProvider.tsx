import { createContext, useContext, FC, ReactNode, useMemo, useState } from 'react';

// types
import { TContext, TPreviewData } from './types';

const PopoverRootContent: React.Context<TContext> = createContext(null);

export const usePopoverRoot = (): TContext => useContext(PopoverRootContent);

export type TPopoverRootProvider = {
  activeOption?: TPreviewData['activeOption'];
  children: ReactNode;
  previewId?: TPreviewData['previewId'];
  selected: boolean;
  setSelected: TContext['setSelected'];
};

export const PopoverRootProvider: FC<TPopoverRootProvider> = ({
  activeOption: initialActiveOption = '',
  children,
  previewId: initialPreviewId = '',
  selected,
  setSelected,
}) => {
  const [activeOption, setActiveOption] = useState(initialActiveOption);
  const [previewId, setPreviewId] = useState(initialPreviewId);

  const setPreviewIdHandler = (
    activeOption: TPreviewData['activeOption'],
    previewId: TPreviewData['previewId'],
  ): void => {
    setActiveOption(activeOption);
    setPreviewId(previewId);
  };

  const value = useMemo(
    () => ({
      activeOption,
      previewId,
      selected,
      setActiveOption,
      setPreviewIdHandler,
      setSelected,
    }),
    [activeOption, previewId, selected],
  );

  return <PopoverRootContent.Provider value={value}>{children}</PopoverRootContent.Provider>;
};
