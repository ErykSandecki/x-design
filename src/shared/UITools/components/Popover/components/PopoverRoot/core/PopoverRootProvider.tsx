import { createContext, useContext, FC, ReactNode, useMemo } from 'react';

// types
import { TContext } from './types';

const PopoverRootContent: React.Context<TContext> = createContext(null);

export const usePopoverRoot = (): TContext => useContext(PopoverRootContent);

export type TPopoverRootProvider = {
  children: ReactNode;
  setSelected: TContext['setSelected'];
};

export const PopoverRootProvider: FC<TPopoverRootProvider> = ({
  children,
  setSelected,
}) => {
  const value = useMemo(
    () =>
      ({
        setSelected,
      }) as TContext,
    [],
  );

  return (
    <PopoverRootContent.Provider value={value}>
      {children}
    </PopoverRootContent.Provider>
  );
};
