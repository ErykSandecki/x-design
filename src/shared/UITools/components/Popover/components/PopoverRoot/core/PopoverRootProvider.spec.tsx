import { noop } from 'lodash';
import { render } from '@testing-library/react';

// core
import { PopoverRootProvider } from './PopoverRootProvider';

describe('PopoverRootProvider snapshots', () => {
  it('should redner PopoverRootProvider', () => {
    // before
    const { asFragment } = render(
      <PopoverRootProvider setSelected={noop}>children</PopoverRootProvider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});
