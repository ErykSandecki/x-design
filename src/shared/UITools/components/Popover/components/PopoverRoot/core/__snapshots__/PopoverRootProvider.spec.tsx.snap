// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopoverRootProvider snapshots should redner PopoverRootProvider 1`] = `
<DocumentFragment>
  children
</DocumentFragment>
`;
